source-pendo/source_pendo/models.py

```diff
@@ -0,0 +1,116 @@
+from estuary_cdk.capture.common import (
+    BaseDocument,
+    ResourceConfig,
+    ResourceState,
+)
+from estuary_cdk.capture.common import (
+    ConnectorState as GenericConnectorState,
+)
+import pendulum
+from pydantic import AwareDatetime, BaseModel, Field
+
+
+def default_start_date():
+    return pendulum.now().subtract(days=1).in_timezone("UTC").format("YYYY-MM-DDTHH:mm:ssZ")
+
+
+class EndpointConfig(BaseModel):
+    startDate: str = Field(
+        description="UTC date and time in the format YYYY-MM-DDTHH:MM:SSZ. Any event data generated before this date will not be replicated. If left blank, the start date will be set to one hour before the present date.",
+        title="Start Date",
+        examples= ["2024-08-27T00:00:00Z"],
+        pattern= "^[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}Z$",
+        default_factory=default_start_date,
+    )
+    apiToken: str = Field(
+        description="API Token to use for Authentication",
+        title="API Token",
+        json_schema_extra={"secret": True},
+    )
+
+
+ConnectorState = GenericConnectorState[ResourceState]
+
+
+class Resource(BaseDocument, extra="allow"):
+    pass
+
+
+class Metadata(BaseDocument, extra="allow"):
+    pass
+
+
+# Event streams have slightly different required fields, so we have models for each event type.
+class PageEvent(BaseDocument, extra="allow"):
+    appId: int
+    hour: int
+    remoteIp: str
+    lastTime: int
```

Comment:
We should have these be actual datetime strings rather than millisecond ints. That's more useful for the consumer, but will complicate the connector code a little bit. For reference, checkout of how `EmailEvent` works in source-hobspot-native, [here](https://github.com/estuary/connectors/blob/da81a628064499560ebca50e828976f89650e3aa/source-hubspot-native/source_hubspot_native/models.py#L373) and [here](https://github.com/estuary/connectors/blob/da81a628064499560ebca50e828976f89650e3aa/source-hubspot-native/source_hubspot_native/api.py#L513-L518) in particular.

Reply:
Thanks for the references, those were helpful! I've changed these int timestamps over to datetime strings. It's interesting how Pydantic handles converting the millisecond int properties in the response to datetime strings with just the `model_validate_json` method.